instrumentation/aws_sdk/lib/opentelemetry/instrumentation/aws_sdk/patches/telemetry.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module AwsSdk
+      module Patches
+        # Patch for Telemetry Plugin Handler in V3 SDK
+        module Handler
+          def call(context)
+            span_wrapper(context) { @handler.call(context) }
+          end
+
+          private
+
+          def span_wrapper(context, &block)
+            service_id = service_id(context)
+            client_method = client_method(service_id, context)
+            context.tracer.in_span(
+              span_name(context, client_method, service_id),
+              attributes: attributes(context, client_method, service_id),
+              kind: span_kind(client_method, service_id)
+            ) do |span|
+              if instrumentation_config[:inject_messaging_context] &&
+                 %w[SQS SNS].include?(service_id)
+                MessagingHelper.inject_context(context, client_method)
+              end
+
+              if instrumentation_config[:suppress_internal_instrumentation]
+                OpenTelemetry::Common::Utilities.untraced { super }
+              else
+                yield span
+              end
+            end
+          end
+
+          def instrumentation_config
+            AwsSdk::Instrumentation.instance.config
+          end
+
+          def service_id(context)
+            context.config.api.metadata['serviceId'] ||
+              context.config.api.metadata['serviceAbbreviation'] ||
+              context.config.api.metadata['serviceFullName']
+          end
+
+          def client_method(service_id, context)
+            "#{service_id}.#{context.operation.name}".delete(' ')
+          end
+
+          def attributes(context, client_method, service_id)
```

Comment:
Is there a way we can share this code between the legacy (ie, pre-sdk-observability support) and here?  Would be good to avoid any duplication of this and make it easier to add future attributes in only one spot.

Reply:
Good idea. I could create a module, like `HandlerHelper`(?) that contains shared methods. Bit of a broader name incase we want to include any methods relating to metrics or logging in the future.

Reply:
Yeah - I like the new HandlerHelper - I think that cleans it up well!